libgobuster/dir.go

```diff
@@ -43,27 +47,40 @@ func (rh *RedirectHandler) RoundTrip(req *http.Request) (resp *http.Response, er
 	return resp, err
 }
 
-// Make a request to the given URL.
-func MakeRequest(s *State, fullUrl, cookie string) (*int, *int64) {
-	req, err := http.NewRequest("GET", fullUrl, nil)
+// MakeRequest ... Make a request to the given URL
+func MakeRequest(s *State, fullURL, cookie string) (*int, *int64) {
+
+	request, err := http.NewRequest(s.Verb, fullURL, strings.NewReader(s.Body))
 
 	if err != nil {
-		return nil, nil
+		os.Exit(1)
```

Comment:
Not a fan of this as a pattern, ideally all errors should be passed up higher, then handled at the top. Exits/panics deep down make life hard.

I believe this has already been solved as part of https://github.com/OJ/gobuster/pull/56 (still WIP)